pmkoalas/models/pnfrag.py

```diff
@@ -0,0 +1,344 @@
+        return transition
+
+    def weightedValueTransition(self):
+        transition = None
+        self.nextToken()
+        label = ''
+        tranId = None
+        silentTran = False
+        weight = 1.0
+        if self.lookahead.tokenInfo == TokenInfo.LABEL:
+            label = self.tranLabel()
+            self.nextToken()
+        else:
+            if self.lookahead.tokenInfo == TokenInfo.SILENT_LABEL:
+                label = self.tranLabel()
+                silentTran = True
+                self.nextToken()
+            else:
+                raise ParseException(
+                        f"Expected label, but found {self.lookahead.tokenStr}")
+
+        if self.lookahead.tokenInfo == TokenInfo.ID_PREFIX:
+            transition = None
+            self.nextToken()
+            tranId = self.id()
+            self.nextToken()
+        if self.lookahead.tokenInfo == TokenInfo.WEIGHT:
+            weight = self.weight()
+            self.nextToken()
+        if self.lookahead.tokenInfo == TokenInfo.WEIGHTED_TRAN_END:
+            transition = self.checkExistingTransition(label)
+        else:
+            self.tokenError(TokenInfo.WEIGHTED_TRAN_END,
+                            TokenInfo.ID_PREFIX)
+        if not transition:
+            if tranId:
+                transition = Transition(label,tid=tranId,silent=silentTran,
+                                    weight=weight)
+                self.idLookup[tranId] = transition
+            else:
+                transition = Transition(label,tid=self.nextId(),
+                                         silent=silentTran, weight=weight)
+                self.labelLookup[label] = transition
+            self.net.add_transition(transition)
+        return transition
+
+
+    def tranLabel(self):
+        return self.lookahead.tokenStr
+
+    def placeLedSubnet(self):
+        head = self.place()
+        if self.lookahead.tokenInfo == TokenInfo.EDGE:
+            self.edge()
+            tran = self.transition()
+            self.edge()
+            tail = self.placeLedSubnet()
+            self.readd_arc(head,tran)
+            self.readd_arc(tran,tail)
+        return head
+
+    def id(self):
+        return int(self.lookahead.tokenStr)
+
+    def weight(self):
+        return float(self.lookahead.tokenStr)
+
+    def tokenError(self,*tokens):
+        raise ParseException(
+                f"Expected one of {tokens} but found {self.lookahead.tokenStr}")
+
+
+    def nextToken(self):
+        self.tokens.pop(0)
+        # at the end of input we return an epsilon token
+        if not self.tokens:
+            self.lookahead = Token(TokenInfo.TERMINAL, "")
+        else:
+            self.lookahead = self.tokens[0]
+
+    def readd_arc(self,fromNode,toNode):
+        self.net.add_arc_between(fromNode,toNode)
+
+    def nextId(self,newfloor=None):
+        if newfloor:
+            self.ctid = max(newfloor,ctid)
+        self.ctid += 1
+        return self.ctid
```

Comment:
Is this not a bug? `ctid` in this context is undefined, it has not been set at all. I can see how `self.ctid` could be not but the non-self reference on 341 in the function call.

Reply:
Looks like dead code, probably how the bug persisted. Have fixed in latest commit.